### ISSUE TITLE
PrefixFilter should set start/end key

### ISSUE THREAD

**Issue by sduskis** [GoogleCloudPlatform/cloud-bigtable-client#1087]
In the HBase API, there's scan.setRowPrefixFilter() for efficient range queries.  There's also PrefixFilter for interesting row filter operations (if the row is has a rowkey in range A, use column A', and if the rowkey is in range B, use column B').

Users use PrefixFilter when they should be using setRowPrefixFilter.  Optimize the ReadRowRequest to use start/end key for cases where this is possible.

**Comment by rayokota** [comment]
If you change the behavior of PrefixFilter, please be sure to account for PrefixFilter in a FilterList with MUST_PASS_ONE

```
FilterList list = new FilterList(FilterList.Operator.MUST_PASS_ONE);
list.add(new PrefixFilter(...));
list.add(new PrefixFilter(...));
scan.setFilter(list);
```

**Comment by igorbernstein2** [comment]
Fixed in #1237
